checkov/secrets/runner.py

```diff
@@ -241,13 +242,16 @@ def search_for_suppression(
             bc_check_id: str,
             severity: Severity | None,
             secret: PotentialSecret,
-            runner_filter: RunnerFilter
+            runner_filter: RunnerFilter,
+            root_folder: str | None
     ) -> _CheckResult | None:
-        if not runner_filter.should_run_check(check_id=check_id, bc_check_id=bc_check_id, severity=severity, report_type=CheckType.SECRETS) and check_id in CHECK_ID_TO_SECRET_TYPE.keys():
+        if (not runner_filter.should_run_check(check_id=check_id, bc_check_id=bc_check_id, severity=severity, report_type=CheckType.SECRETS) and check_id in CHECK_ID_TO_SECRET_TYPE.keys()) or \
+                not runner_filter.should_run_check_for_file(check_id=check_id, file_full_path=secret.filename, root_folder=root_folder):
```

Comment:
This filter should be applied to `should_run_check`, runner should not know the internals of all the possible skips. You can have a default value of `None` to keep backwards compatibility